cli/lib/kontena/cli/stacks/yaml/reader.rb

```diff
@@ -366,7 +378,7 @@ def interpolate(content, use_opto: true, substitutions: {}, raise_on_unknown: fa
           if row.strip.start_with?('interpolate:') || row.strip.start_with?('evaluate:')
             row
           else
-            row.gsub(/(?<!\$)\$(?!\$)\{?\w+\}?/) do |v| # searches $VAR and ${VAR} and not $$VAR
+            row.gsub(/(?<!\$)\$(?!\$)\{?(?:\w|\.)+\}?/) do |v| # searches $VAR and ${VAR} and not $$VAR
```

Comment:
What happens to `foo=$GRID.kontena.local`? Would these interpolations have worked earlier, but would now fail?

Reply:
Yep, those would have worked but after this will not. Perhaps `$foo[foo]` / `${foo[foo]}` would be safer. Also not entirely sure what happens in liquid with this dot notation.

Implementing groups in opto would probably help a bit. https://github.com/kontena/opto/issues/39

Reply:
One option could be to only allow this in Liquid.